xla/service/gpu/BUILD

```diff
@@ -1345,29 +1345,6 @@ cc_library(
     ],
 )
 
-cc_library(
```

Comment:
Any particular reason why you decided to re-inline the pipeline in `gpu_compiler.cc`? That file is pretty massive, and the pipeline is standalone. How about we just keep it in a separate file?

Reply:
I see no other uses of this pipeline anywhere. Now it takes 2 extra files and an empty method GpuCompiler::PrepareHloModuleForIrEmitting(). IMO moving it to the compiler class improves readability, but I wanted to hear your reasons to outline it.

Reply:
In general, I think moving pipelines to dedicated files makes sense, simply because they're meant to delineate well-scoped, independent combinations of passes. Ideally, we should also have tests for them that ensure that they perform the adequate combinations of transformations---which is easier to do if they have their own file as well.

The main reason for outlining this particular one in the first place, though, was to prepare for changes that would involve calling this particular pipeline from within other (autotuning) passes, though. While it wasn't done in the end, keeping pipelines in `gpu_compiler.cc` also means that the functionality can not be reused in passes without creating circular dependencies.